src/board_controller/brainalive/brainalive.cpp

```diff
@@ -371,21 +375,36 @@ void BrainAlive::adapter_1_on_scan_found (
 void BrainAlive::read_data (simpleble_uuid_t service, simpleble_uuid_t characteristic,
     uint8_t *data, size_t size, int channel_num)
 {
-    if (size != BRAINALIVE_PACKET_SIZE)
+    if (size == brainalive_packet_size)
     {
-        safe_logger (spdlog::level::warn, "unknown size of BrainAlive Data {}", size);
-        return;
-    }
-
-    for (int i = 0; i < (int)size; i += 32)
-    {
-        double eeg_data[9] = {0};
-        for (int j = i + 4, k = 0; j < i + 28; j += 3, k++)
+        for (int i = 0; i < (int)size; i += brainalive_single_packet_size)
         {
-            eeg_data[k] = (((data[j] << 16 | data[j + 1] << 8 | data[j + 2]) << 8) >> 8) *
-                BRAINALIVE_EEG_SCALE_FACTOR / BRAINALIVE_EEG_GAIN_VALUE;
+            double ba_data[ba_brainflow_package_size] = {0};
```

Comment:
as discussed earlier it probably makes sense to determine it fron board_descr, and to make it future proof you can use it for package_num_channel, eeg_channels, etc. Here is how it looks like: https://github.com/brainflow-dev/brainflow/blob/c9ae582ca10aa2e2a9eb7ea3e30c71c64d40c001/src/board_controller/freeeeg/freeeeg.cpp#L171 https://github.com/brainflow-dev/brainflow/blob/c9ae582ca10aa2e2a9eb7ea3e30c71c64d40c001/src/board_controller/freeeeg/freeeeg.cpp#L145 this way next time when you change brainflow_boards.cpp you will not need to syncup this array here